### PR TITLE
Add scene filtering sample

### PR DESCRIPTION
**Pull request by tanneryould**
# Description

Adds the "Filter features in scene" sample

## Type of change

- [x] New sample implementation

**Platforms tested on**:

- [x] macOS

## Checklist

- [x] Runs and compiles on all active platforms as a standalone sample
- [x] Runs and compiles in the sample viewer(s)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [x] If adding a new sample, it is added to the sample viewer

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  connect(m_detailedBuildingsSceneLayer, &ArcGISSceneLayer::doneLoading, this, [this](const Error error)
+  {
+    if (!error.isEmpty())
+    {
+      qWarning() << "San Fransisco Buildings Layer failed to load." << error.message() << error.additionalMessage();
+      return;
+    }
+
+    // Construct a red polygon that shows the extent of the detailed buildings scene layer
+    Envelope sfExtent = m_detailedBuildingsSceneLayer->fullExtent();
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp:81]: could be marked const

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+void FilterFeaturesInScene::loadScene()
+{
+  // Show the detailed buildings scene layer and the extent graphic
+  m_scene->operationalLayers()->append(m_detailedBuildingsSceneLayer);
+  m_sceneView->graphicsOverlays()->first()->graphics()->append(m_sanFransiscoExtentGraphic);
+}
+
+void FilterFeaturesInScene::filterScene()
+{
+  // Hide buildings within the detailed building extent so they don't clip
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp:123]: what code is this comment referring to?

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/FilterFeaturesInScene/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+You can selectively show or hide features within a scene view using `setFeaturesVisible` to filter specific features or set a `SceneLayerPolygonFilter` to filter features given a polygon. Through these methods, you can directly control what users see within a specific scene view to give a more focused or cleaner user experience.
+
+## How to use the sample
+
+The sample initializes showing the 3D buildings OpenStreetMap layer. Click the "Load detailed buildings" button to load an additional scene layer that contains more detailed buildings. Notice how the two scene layers overlap and clip into each other. Click the "Filter OSM buildings in extent" button, to set a `SceneLayerPolygonFilter` and filter out the OpenStreetMap buildings within the extent of the detailed buildings scene. Notice how the OSM buildings within and intersecting the extent of the detailed buildings layer are hidden. Click the "Reset scene" button to hide the detailed buildings scene layer and clear the OSM buildings filter.
+
+## How it works
+
+1. Construct a `Basemap` for the scene using an OpenStreetMap Topographic `VectorTileLayer` and the OpenStreetMap 3D Buildings `ArcGISSceneLayer` as baselayers.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/FilterFeaturesInScene/README.md:17]: if we put it in backticks, we should use the full proper name. If we want to be casual, can just remove the backticks
```suggestion
1. Construct a `Basemap` for the scene using an OpenStreetMap Topographic `ArcGISVectorTileLayer` and the OpenStreetMap 3D Buildings `ArcGISSceneLayer` as baselayers.
```

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/FilterFeaturesInScene/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## About the data
+
+This sample uses the [OpenStreetMap 3D Buildings](https://www.arcgis.com/home/item.html?id=ca0470dbbddb4db28bad74ed39949e25) which provides generic 3D outlines of buildings throughout the world. It is based on the OSM Daylight map distribution and is hosted by Esri. It uses the [San Fransisco 3D Buildings](https://www.arcgis.com/home/item.html?id=d3344ba99c3f4efaa909ccfbcc052ed5) scene layer which provides detailed 3D models of buildings in San Fransisco, California, USA.
+
+## Additional information
+
+This sample uses `SceneLayerPolygonFilterSpatialRelationship::Disjoint` to hide all features within the extent of the given geometry. You can alternatively use `SceneLayerPolygonFilterSpatialRelationship::Contains` to only show features within the extent of the geometry. 
+
+You can also show or hide features in a scene layer using `ArcGISSceneLayer::setFeatureVisible` or `setFeaturesVisible` and pass in a feature or list of features and a boolean value to set their visibility.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/FilterFeaturesInScene/README.md:37]:
```suggestion
You can also show or hide features in a scene layer using `ArcGISSceneLayer.setFeatureVisible` or `setFeaturesVisible` and pass in a feature or list of features and a boolean value to set their visibility.
```

**Review by ldanzinger — APPROVED** [review]
LGTM, just a couple of small comments

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp (hunk 2 above)
Comment (tanneryould) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp:123]: It's referring to the code that occurs in the method - either the `if` or `else` blocks. It's intended to be kind of like a header. We don't normally put comments before the method (ie line 120) so I placed it there.

**Pull request merged**